torch/optim/adam.py

```diff
@@ -373,11 +378,17 @@ def _single_tensor_adam(params: List[Tensor],
                 (param.is_cuda and step_t.is_cuda) or (param.is_xla and step_t.is_xla)
             ), "If capturable=True, params and state_steps must be CUDA or XLA tensors."
 
-        # update step
-        step_t += 1
+        if not decoupled_weight_decay:
```

Comment:
decoupled_weight_decay should only matter when weight_decay is nonzero, right?

Reply:
could probably rearrange the code with that observation

Reply:
I switched it up but nested the conditionals so the call to mul_() isn't made unnecessarily. In the case weight_decay is 0 the call to mul_ won't change anything but I figured it might be expensive, let me know if you'd prefer something like:

```python
if weight_decay != 0 and if not decoupled_weight_decay:
    grad = grad.add(param, alpha=weight_decay)
else:
    # Perform stepweight decay
    param.mul_(1 - lr * weight_decay)
```